sdk/servicebus/Azure.Messaging.ServiceBus/src/ServiceBusException.cs

```diff
@@ -55,74 +55,19 @@ public override string Message
-                                  string entityName,
-                                  FailureReason reason) : this(isTransient, entityName, null, reason, null)
-        {
-        }
-
-        /// <summary>
-        ///   Initializes a new instance of the <see cref="ServiceBusException"/> class.
-        /// </summary>
-        ///
-        /// <param name="isTransient"><c>true</c> if the exception should be considered transient; otherwise, <c>false</c>.</param>
-        /// <param name="entityName">The name of the Service Bus entity to which the exception is associated.</param>
-        /// <param name="message">The error message that explains the reason for the exception.</param>
-        ///
-        public ServiceBusException(bool isTransient,
-                                  string entityName,
-                                  string message) : this(isTransient, entityName, message, FailureReason.GeneralError, null)
-        {
-        }
-
-        /// <summary>
-        ///   Initializes a new instance of the <see cref="ServiceBusException"/> class.
-        /// </summary>
-        ///
-        /// <param name="isTransient"><c>true</c> if the exception should be considered transient; otherwise, <c>false</c>.</param>
-        /// <param name="entityName">The name of the Service Bus entity to which the exception is associated.</param>
-        /// <param name="message">The error message that explains the reason for the exception.</param>
-        /// <param name="reason">The reason for the failure that resulted in the exception.</param>
-        ///
-        public ServiceBusException(bool isTransient,
-                                  string entityName,
-                                  string message,
-                                  FailureReason reason) : this(isTransient, entityName, message, reason, null)
-        {
-        }
-
         /// <summary>
         ///   Initializes a new instance of the <see cref="ServiceBusException"/> class, using the <paramref name="reason"/>
         ///   to detect whether or not it should be transient.
         /// </summary>
         ///
-        /// <param name="entityName">The name of the Service Bus entity to which the exception is associated.</param>
         /// <param name="message">The error message that explains the reason for the exception.</param>
         /// <param name="reason">The reason for the failure that resulted in the exception.</param>
+        /// <param name="entityName">The name of the Service Bus entity to which the exception is associated.</param>
         ///
-        public ServiceBusException(string entityName,
-                                  string message,
-                                  FailureReason reason) : this(default, entityName, message, reason, null)
+        public ServiceBusException(
+            string message,
+            FailureReason reason,
+            string entityName = default) : this(default, entityName, message, reason, null)
```

Comment:
The parameters passed to the constructor seem jumbled up, it should be -  `this(default, message, entityName, reason, null)`

Reply:
Fixed!